zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -250,35 +261,20 @@ where
         // Since this is a very large RPC, we use separate functions for each group of fields.
         async move {
             let _tip_height = best_chain_tip_height(&latest_chain_tip)?;
+            let mempool_txs = select_mempool_transactions(mempool).await?;
 
-            // TODO: put this in a separate get_mempool_transactions() function
-            let request = mempool::Request::FullTransactions;
-            let response = mempool.oneshot(request).await.map_err(|error| Error {
-                code: ErrorCode::ServerError(0),
-                message: error.to_string(),
-                data: None,
-            })?;
-
-            let transactions = if let mempool::Response::FullTransactions(transactions) = response {
-                // TODO: select transactions according to ZIP-317 (#5473)
-                transactions
-            } else {
-                unreachable!("unmatched response to a mempool::FullTransactions request");
-            };
+            let miner_fee = miner_fee(&mempool_txs);
 
-            let merkle_root;
-            let auth_data_root;
+            /*
+            TODO: create a method Transaction::new_v5_coinbase(network, tip_height, miner_fee);
+            */
+            let coinbase_tx = mempool_txs[0].transaction.clone();
```

Comment:
Should we run the `coinbase_tx` through the `transaction::Verifier`?

We'll need a new request type which returns a coinbase result, because the `remaining_value_balance()` will be negative. (Maybe this can go in another ticket?)

Reply:
It's probably not strictly necessary since the generated coinbase is a function of verified transactions and it's being run through the `transaction::Verifier` by the `BlockVerifier` when the block is submitted, but it might be nice for testing and debugging.

Reply:
I'll see what I can do in the next PR.